r/tests/testthat/test-RecordBatch.R

```diff
@@ -137,7 +137,6 @@ test_that("read_record_batch handles various streams (ARROW-3450, ARROW-3505)",
   batch4 <- read_record_batch(mmap_file)
   batch5 <- read_record_batch(bytes)
   batch6 <- read_record_batch(buf_reader)
-  expect_error(read_record_batch(buf_reader))
```

Comment:
@romainfrancois I believe this PR is now working! However, I want to talk about this test.

I had to remove this to get `R CMD CHECK` or `devtools::test()` to complete. I _think_ the issue is that `expect_error` cannot trap C++ exceptions. When running with this test restored, testing immediately stops with error:

```
terminate called after throwing an instance of 'std::bad_alloc'
    what():  std::bad_alloc
  Aborted
```

Have you ever seen that issue with `expect_error()` before? I tried googling around and nothing obvious jumped out to me. I have no idea how this could be working on Travis but not in my setup on docker :/.

Reply:
Created issue for it: https://issues.apache.org/jira/browse/ARROW-3833

Reply:
That’s what Rcpp::stop does, but this is supposed to be caught by the generated code.